src/ir/subst.rs

```diff
@@ -18,6 +20,10 @@ where
             .iter()
             .find_map(|(k, v)| if k == key { Some(v) } else { None })
     }
+
+    pub fn insert(&mut self, key: K, value: V) {
+        self.0.push((key, value));
```

Comment:
I'm not sure if we should be mutably updating bindings. Or at least, exposing this as the function signature. It might be better to consume the binding by making the function take `self` so that it's explicit that the old binding is now truly gone

Reply:
What would this look like?